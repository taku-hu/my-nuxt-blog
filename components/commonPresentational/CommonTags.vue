<template>
  <ul class="flex flex-wrap justify-center">
    <li v-for="tag in tags" :key="tag.fields.name" class="m-2">
      <nuxt-link
        class="bg-blue-400 rounded shadow-sm text-xs font-bold text-white p-2"
        :to="`/tag/${tag.sys.id}/${tag.fields.name}`"
      >
        <fa :icon="faTags" />
        {{ tag.fields.name }}
      </nuxt-link>
    </li>
  </ul>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Entry, Field } from 'contentful/index'
import { faTags } from '@fortawesome/free-solid-svg-icons'

export default Vue.extend({
  props: {
    tags: {
      type: Array as PropType<Entry<Field>[]>,
      required: true
    }
  },
  computed: {
    // static
    faTags: () => faTags
  }
})
</script>
