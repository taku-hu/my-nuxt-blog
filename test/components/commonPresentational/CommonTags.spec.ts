import { shallowMount } from '@vue/test-utils'
// import { faTags } from '@fortawesome/free-solid-svg-icons'
// @ts-ignore
import CommonTags from '@/components/commonPresentational/CommonTags.vue'

describe('CommonHomeCard.vue', () => {
  const propsData = {
    tags: [
      {
        fields: {
          name: 'name1'
        },
        sys: {
          id: '12345'
        }
      },
      {
        fields: {
          name: 'name2'
        },
        sys: {
          id: '54321'
        }
      }
    ]
  }

  const wrapper = shallowMount(CommonTags, {
    propsData
  })
  it('props', () => {
    expect(wrapper.props()).toEqual(propsData)
  })
  // it('template', () => {
  //   expect(wrapper.html()).toContain(`<a class="bg-blue-400 rounded shadow-sm text-xs font-bold text-white p-2"><span icon="${faTags}"></span>name1</a>`)
  // })
})
