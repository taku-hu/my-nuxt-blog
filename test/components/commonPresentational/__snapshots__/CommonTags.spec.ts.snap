// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonHomeCard.vue snapshot 1`] = `
<ul
  class="flex flex-wrap justify-center"
>
  <li
    class="m-2"
  >
    <a
      class="bg-blue-400 rounded shadow-sm text-xs font-bold text-white p-2"
    >
      <span
        icon="[object Object]"
      />
      
      name1
    
    </a>
  </li>
  <li
    class="m-2"
  >
    <a
      class="bg-blue-400 rounded shadow-sm text-xs font-bold text-white p-2"
    >
      <span
        icon="[object Object]"
      />
      
      name2
    
    </a>
  </li>
</ul>
`;
